### PR TITLE
Enable power users to bypass device_map="auto" training block

### PR DESCRIPTION
**Pull request by muellerzr**
This PR fulfills https://github.com/huggingface/accelerate/issues/1368#issuecomment-1689817806 by letting users set `ACCELERATE_BYPASS_DEVICE_MAP="true"` as an environmental variable so that if users wish to try an experimental approach at TP as described in the issue thread, they can.

### CODE REVIEW

**Comment by muellerzr** [comment]
cc @maxidl :)

File: src/accelerate/accelerator.py [hunk 1]
```diff
@@ -1328,7 +1330,12 @@ def prepare_model(self, model: torch.nn.Module, device_placement: bool = None, e
             device_placement = self.device_placement and self.distributed_type != DistributedType.FSDP
         self._models.append(model)
 
-        if self.verify_device_map(model) and self.distributed_type != DistributedType.NO:
+        # TODO: Look at enabling native TP training directly with a proper config
+        if (
+            self.verify_device_map(model)
+            and self.distributed_type != DistributedType.NO
+            and os.environ.get("ACCELERATE_BYPASS_AUTO", "false") != "true"
```
Comment (SunMarc) [inline src/accelerate/accelerator.py:1337]: Maybe change to something like `ACCELERATE_BYPASS_DEVICE_MAP` ?

**Review by SunMarc — APPROVED** [review]
LGTM ! Just one little nit.

**Review by sgugger — APPROVED** [review]
Thanks for adding this! LGTM with Marc's comment on the name (agreed his is better)

**Pull request merged**